dotrix_core/src/renderer.rs

```diff
@@ -169,6 +176,13 @@ pub fn startup(mut renderer: Mut<Renderer>, mut globals: Mut<Globals>, window: M
 
 /// Frame binding system
 pub fn bind(mut renderer: Mut<Renderer>, mut assets: Mut<Assets>) {
+    // Clear any dropped shader's pipelines
+    for id in renderer.backend().get_pipeline_shaders() {
+        if assets.get::<Shader>(id).is_none() {
+            renderer.drop_shader_pipeline(id);
+        }
+    }
+
```

Comment:
I don't think this operation should be performed on every frame. I suggest to move it bellow the `let mut loaded = true;` and do not drop all pipelines in the cycle bellow, but add a method instead `renderer.drop_all_pipelines()`

in that case `bind` system won´t perform all this checks when nothing changed.

Reply:
I put this here to handle the case when someone calls:

```rust
assets.remove::<Shader>(some_shader_id)
```

The underlying pipeline is never cleaned up

Reply:
I am trying to think about the use case. This is a rare operation, isn´t it?

Reply:
Yes quite rare, I was just thinking of edge cases.

Perhaps this needs something more complex on the asset stage. Have some way in assets to mark that `hey this was deleted/added/changed this frame` and let a system handle it